### PR TITLE
Migrate Seesaw from .Net IoT to .NET nanoFramework

### PR DESCRIPTION
**Pull request by usmanatron**
## Description

* Migrate Seesaw package from .Net IoT to nanoFramework.  Followed the same principles as with the other devices.
* Minimal changes were needed to the actual code, though I tidied up the samples to make them easier to work with.
* Most interesting change is the GpioDriver - I don't think it's needed anymore so I removed it.

## Motivation and Context

- Resolves nanoFramework/Home#862

## How Has This Been Tested?
I don't have a Seesaw to test against.  Everything compiles!

## Screenshots
N/A

## Types of changes
- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by usmanatron** [comment]
Azure Build is failing due to a NullRef when checking the nuspec.  I thought that was because the versions were wrong but it still seems unhappy.

File: devices/Seesaw/app.config [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (Ellerbach) [inline devices/Seesaw/app.config:1]: this should be removed. We don't need app.config. They are creating issues.

File: devices/Seesaw/Seesaw.nfproj [hunk 2]
```diff
@@ -20,55 +20,53 @@
...
+    </Reference>
+    <Reference Include="System.Device.I2c, Version=1.0.2.3, Culture=neutral, PublicKeyToken=c07d481e9758c731">
+      <HintPath>packages\nanoFramework.System.Device.I2c.1.0.2\lib\System.Device.I2c.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Diagnostics.Stopwatch, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null">
+      <HintPath>packages\nanoFramework.System.Diagnostics.Stopwatch.1.0.259\lib\System.Diagnostics.Stopwatch.dll</HintPath>
     </Reference>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (Ellerbach) [inline devices/Seesaw/Seesaw.nfproj:47]: remove the app.config from the nfproj as well ;-) Otherwise it won't build properly

Comment (usmanatron) [reply devices/Seesaw/Seesaw.nfproj]: Yep, noticed that as soon as I committed!

File: devices/Seesaw/Seesaw.nuspec [hunk 3]
```diff
@@ -20,7 +20,13 @@
     <summary>Iot.Device.Seesaw assembly for .NET nanoFramework C# projects</summary>
     <tags>nanoFramework C# csharp netmf netnf Iot.Device.Seesaw</tags>
     <dependencies>
-      <dependency id="nanoFramework.CoreLibrary" version="1.10.4-preview.11" />
+      <dependency id="nanoFramework.CoreLibrary" version="1.11.7" />
+      <dependency id="nanoFramework.Runtime.Events" version="1.9.2" />
```
Comment (Ellerbach) [inline devices/Seesaw/Seesaw.nuspec:24]: are all those the latest versions? and correctly in sync with the nfproj? Just not super easy to read in this mode.

File: devices/Seesaw/SpanPinValuePair.cs [hunk 4]
```diff
@@ -1,9 +1,12 @@
-namespace System
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.Seesaw
```
Comment (Ellerbach) [inline devices/Seesaw/SpanPinValuePair.cs:4]: where is this used? I don't think it is at all. That's related to the GPIO driver (which is not here and not useful) and I don't think that's needed.

File: devices/Seesaw/README.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 Adafruit Seesaw is a near-universal converter framework which allows you to add and extend hardware support to any I2C-capable microcontroller or microcomputer. Instead of getting separate I2C GPIO expanders, ADCs, PWM drivers, etc, seesaw can be configured to give a wide range of capabilities.
 
-This binding provides an Api which is close to the one provided by Adafruit themselves but also implements IGpioController so that available Gpio pins can be used in place of any 'on board' ones but using the standard IoT API.
+This binding provides an API which is close to the one provided by Adafruit themselves but also implements IGpioController so that available Gpio pins can be used in place of any 'on board' ones but using the standard IoT API.
```
Comment (Ellerbach) [inline devices/Seesaw/README.md:5]: I can't check the fritzing diagram and the associated images. If it's with a Raspberry, then please adjust for an ESP32 or equivalent.

**Review by Ellerbach — COMMENTED** [review]
All up, looks good. Not easy to review the files as you moved a lot of them. So can't check if any generics stays there and things like this. And doesn't make it easy for the all up review!
Have you tested it on a real hardware?

**Comment by josesimoes** [comment]
@usmanatron nanovc fixed and build running again.

(you'll notice that some dependencies have their IDs wrong in nuspec)

**Comment by josesimoes** [comment]
@usmanatron can you please fix the merge conflicts that popped?

**Comment by josesimoes** [comment]
@usmanatron not wanting to push you but doing it... 😁 any chance to get this done so it can be merged?

**Comment by josesimoes** [comment]
@usmanatron this is getting behind... will you be handling this and bring it up to speed or should we take over?

**Comment by networkfusion** [comment]
@usmanatron any update?

**Comment by Ellerbach** [comment]
@usmanatron will you have the time to continue the work? The branch is quite behind the current one and you still have to address some of the comments.
Thanks!

**Comment by Ellerbach** [comment]
@usmanatron any update on this work started in January?

**Comment by josesimoes** [comment]
hey @usmanatron any plans to resume work on this? 😃

**Comment by networkfusion** [comment]
Given no update in 2 years, I am going to close this PR as superseeded with #1036.

**Pull request closed without merging**